src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -144,17 +147,32 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
 
 async def rclone_check_progress(job, proc):
+    dropbox__restricted_content = False
     while True:
         read = (await proc.stdout.readline()).decode()
-        job.logs_fd.write(read.encode("utf-8", "ignore"))
         if read == "":
             break
+        if "failed to open source object: path/restricted_content/" in read:
+            job.internal_data["dropbox__restricted_content"] = True
+            dropbox__restricted_content = True
+        job.logs_fd.write(read.encode("utf-8", "ignore"))
         reg = RE_TRANSF.search(read)
         if reg:
             transferred = reg.group(1).strip()
             if not transferred.isdigit():
                 job.set_progress(None, transferred)
 
+    if dropbox__restricted_content:
+        message = "\n" + (
```

Comment:
There's some extra details in this message that can be trimmed and it might be good to have a couple links to Dropbox ToS and their Support portal:

`The Dropbox sync has failed due to restricted content in the shared folders. Please see the <a href="https://www.dropbox.com/terms" target="_blank">Dropbox Terms of Service</a>. All unrestricted content is synchronized and no files have been deleted. Please see the task logs to identify the restricted files and exclude them from the synchronization. Contact <a href="https://www.dropbox.com/support" target="_blank">Dropbox Support</a> if files are erroneously restricted.`

Reply:
@themylogin what do you think?

Reply:
Oops, I've addressed Aaron review and have not noticed that Tim's review also contains other concerns. I don't think that including link to https://www.dropbox.com/terms will help anyone, it is giant and useless. Contact support link — included.